pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -3166,13 +3166,15 @@ public void updateArray(String columnName, Array x) throws SQLException {
       if (sqlType == Types.NUMERIC || sqlType == Types.DECIMAL) {
         return type.cast(getBigDecimal(columnIndex));
       } else {
-        throw new SQLException("conversion to " + type + " from " + sqlType + " not supported");
+        throw new PSQLException(GT.tr("conversion to {1} from {2} not supported", type, sqlType),
```

Comment:
Are you sure placeholders are 1-based?
We use MessageFormat ( https://github.com/pgjdbc/pgjdbc/blob/246b759cdc264c2732717dbd6ff9f8f472024196/pgjdbc/src/main/java/org/postgresql/util/GT.java#L59 ), and it is 0-based (see https://docs.oracle.com/javase/7/docs/api/java/text/MessageFormat.html )

Reply:
I must have copy and pasted from the wrong place, eg. https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java#L349

Fixed